src/common/detectTerminalShell.c

```diff
@@ -241,6 +240,7 @@ const FFTerminalShellResult* ffDetectTerminalShell(FFinstance* instance)
     char ppid[256];
     snprintf(ppid, 255, "%i", getppid());
     getTerminalShell(&result, ppid);
+    result.terminalExeName = result.terminalExe.chars;
```

Comment:
This is definitely wrong, as it overwrites the `result.terminalExeName` set by `getTerminalShell`:
```c
ffStrbufAppendS(&result->terminalProcessName, name);
getProcessInformation(pid, &result->terminalProcessName, &result->terminalExe, &result->terminalExeName);
```
Which calls this code two methods later:
```c
uint32_t lastSlashIndex = ffStrbufLastIndexC(exe, '/');
if(lastSlashIndex < exe->length)
    *exeName = exe->chars + lastSlashIndex + 1;
```
Most probably your issue is something there.